pkg/version/version.go

```diff
@@ -26,6 +28,12 @@ var (
 	// Set in hack/build.sh.
 	Commit = ""
 
+	// HostArch is the payload architecture for which the installer was built,
```

Comment:
Why do we care about the host architecture? I would expect instead to have something like `defaultReleaseArchitecture`, which is would be populated when extracting the installer from the release image.

Reply:
Don't recall if we discussed such an option, but that would require support in (and therefore a matching version of) `oc`.  I thought we did settle on relying on the build architecture, in that an installer binary built in the containers is going to be part of a payload of the container build architecture, or `GOHOSTARCH`, which is why I named it `HostArch`.

Reply:
I must be misunderstanding something about the build pipeline. Let's take for example that we want an installer that runs on amd64 and installs a cluster running arm64. In that case, will the container from which the build will be performed use the arm64 architecture, since the release image will be targeting arm64? And, then, will the build pipeline be doing a cross-architecture build for producing installers targeting all of the platforms that we support for the installer?

Let's assume that my understanding above is accurate. In that case, you are correct that we would not need the release architecture set during the extraction process. So we are good there. I still don't think that `HostArch` is a great name for this variable, however. That name refers to how the value is determined but not what we want the variable to represent.

I still have concerns about breaking our backwards compatibility by defaulting the target architecture to the host architecture. This may not be very problematic for OCP, but I could see it being problematic for OKD, where users do their own builds more frequently. Can we pass the target architecture as an optional argument to `./hack/build.sh` instead so that the target architecture can be set in the dockerfile?

Reply:
You understand the build pipeline correctly.
I have renamed the variable to `defaultArch`.
Given that your average user does not have access to P&Z, and the installer was only recently made buildable on arm64, and any other architecture would still default to amd64, what is the case that you worry about this breaking?

Reply:
My concern is that an install-config.yaml that produced a cluster in OpenShift 4.8 should continue to produce the same cluster in OpenShift 4.9. An install-config.yaml that produced a cluster on my machine should produce the same cluster on your machine, regardless of the architecture of our machines. I thought that was the whole point of this effort, namely to isolate the architecture of the machine running the installer from the architecture of the cluster created.

Reply:
Right now, if you `create install-config` with 4.8 for aws, you get `compute[0].architecture: amd64` and `controlPlane.architecture: amd64`.  There should be no reason that this file would not work after this PR.  The point of this PR is to change what `create install-config` writes to match the payload regardless of where it is *run*, so that even a linux/amd64 or darwin/amd64 installer from an ARM payload (and hence *built* on ARM) will write `arm64` there, and the user doesn't have to modify the install-config even in that use case.

Reply:
It is not clear to me that we have articulated clearly the problem that we are trying to solve here. Perhaps our best path forward here is to exclude this change from the patchset so that we can get arm64 enabled, then followup with an issue and/or call to articulate and demonstrate the issues that we see and are trying to solve with this particular piece of code

Reply:
Right now, when someone downloads an installer from an x86 payload (which could run either on linux or mac) and chooses to install to AWS, the install-config.yaml generated by `create install-config` "just works", iow no modification is *necessary* to the yaml to get a working, default cluster.  The purpose this serves is to provide the very same experience for ARM payloads: that an installer from an ARM payload -- regardless of where it *runs* -- will write an install-config that "just works", without modification, to install that OpenShift payload to ARM on AWS.

Reply:
The `controlPlane.architecture` field is optional. If a user is supplying their own install-config.yaml file, rather than using `create install-config`, the `controlPlane.architecture` field can be omitted. In 4.8, the installer would default the architecture to amd64. If the user then takes the same install-config.yaml, with the field omitted, and does an installation on 4.9, the installer must continue to default to amd64. It cannot default to whatever the host architecture is where the installer is run.

Reply:
So you're asking us to separate the `create install-config` default from the `create cluster` if unspecified default.  OK, @Prashanth684 any ideas on how to do that?

Reply:
As for setting the arch in the Dockerfile, I just pushed a new commit to that end.

Reply:
@yselkowitz i think that investigation warrants a separate discussion as a separate issue/PR. i do not want to pollute this PR with more changes back and forth on this. this is not that trivial a problem to solve and also an important one to get it done right, so i suggest breaking this out.